scipy/ndimage/tests/test_filters.py

```diff
@@ -86,6 +86,30 @@ def test_valid_origins():
         assert_raises(ValueError, filter, data, 3, origin=2)
 
 
+def test_gaussian_kernel_truncate():
+    # Test that Gaussian kernels can be truncated at different widths.
+    num_nonzeros_2 = sndi.calculate_gaussian_kernel(5, truncate=2)
+    assert_equal(len(num_nonzeros_2), 2 * int(2 * 5. + 0.5) + 1)
+    num_nonzeros_5 = sndi.calculate_gaussian_kernel(5, truncate=5)
+    assert_equal(len(num_nonzeros_5), 2 * int(5 * 5. + 0.5) + 1)
+
+
+def test_manual_filter():
+    arr = np.ones([20, 20])
+    sigma = 10
+    truncate = 5
+
+    result1 = sndi.gaussian_filter(arr, sigma, truncate=truncate)
+
+    weights = sndi.calculate_gaussian_kernel(sigma, truncate=truncate)
+
+    output = arr.copy()
```

Comment:
So this little block here is what the intentions of the modifications to `gaussian_filter` were designed to avoid. Basically preventing the user from doing this iteration (which as I say, is not a massive burden).